File: 프로그래머스/unrated/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        int size = p.length();
        
        Long pN = Long.parseLong(p);
        for(int i =0; i<=t.length() - size; i++){
            Long n = Long.parseLong(t.substring(i,i+size));
            if(n<=pN){
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/unrated/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    answer = 0
    size = len(p)
    first_value = p[0]
    i = 0
    while i <= len(t) - size:
        
        if t[i] > first_value:
            i+=1
            continue
        
        
        tmp = t[i:i + size]
        #print(tmp)
        if tmp <= p:
            answer+=1
        
        i+=1
        
    return answer